proposal/color-4-new-spaces-js.d.ts.md

```diff
@@ -218,68 +285,190 @@ colors as best they can, but it may be necessary in some cases.
 toGamut(): SassColor;
 ```
 
-#### `changeChannels`
-
-Returns a new SassColor as the result of changing some of [`internal`]'s
-channels. The `space` value defaults to the `space` of [`internal`], and any
-combination of channels in that space may be changed. Throws an error if any
-`channel` is not present in `space`.
-
-* If `space` is not defined, let `space` be the value of [`space(internal)`].
+#### `change`
 
-* Let `arguments` be the key value pairs in `channels` as keyword arguments.
+Replace the definition of [color.change] with the following:
+
+[color.change]: ../spec/js-api/value/color.d.ts.md#change
 
-* Return the value of the result of [`change(internal, ...arguments)`].
-
-[`space(internal)`]: #space
-
-[`change(internal, ...arguments)`]: ./color-4-new-spaces.md#colorchange
+This algorithm takes a JavaScript object `options` and returns a new SassColor
+as the result of changing some of [`internal`]'s components.
+
+The `space` value defaults to the `space` of [`internal`], and any combination
+of channels and alpha in that space may be changed.
+
+If `space` is not a [legacy color space], a channel value of `null` will result
+in a [missing component][missing components] value for that channel.
+
+* Let `initialSpace` be the value of [`internal.space()`].
+
+* If `options.space` is not defined, let `space` be the value of `initialSpace`.
+
+* Let `keys` be a list of the keys in `options` without `space`.
+
+* Let `components` be `"alpha"` and the channels in `space`.
+
+* If any key in `keys` is not the name of a channel in `components`, throw an
+  error.
+
+* If `space` is not equal to `initialSpace`, let `convertedColor` be the result
+  of [`internal.toSpace(space)`].
+
+* If `space` equals `hsl`, let `changedColor` be the result of:
+
+```js
+  SassColor({
+    hue: options.hue ?? internal.channel('hue'),
```

Comment:
We should probably also emit a deprecation warning if the user explicitly passes `null` for a legacy color. This was historically disallowed by the TypeScript types, but if users were using pure JS it worked the same as `undefined`.

Actually, we could also say that if users pass `space` explicitly, even legacy colors treat `null` as missing.

Reply:
In ab8864f, I emitted the `alpha-null` deprecation.

I'm not sure that we can treat `null` as missing if the change is happening in a legacy space. If a color has an `initialSpace` of `hsl` with an `space` of `hsl` specified, and `saturation: null`, we would end with a color that could not be represented. 

If `initialSpace` is `lch`, but set the saturation to missing in the `hsl` space, the `hsl` intermediary color would not be able to be represented, and before conversion back to `lch`, the `saturation` would be set to `0`.

Because the CSS Color Conversion spec specifies that converting colors with missing channels sets that value to 0, I updated the specification to only convert to `initialSpace` at the end if `space` is not equal to `initialSpace`.

Reply:
I'm not sure what you mean by "can't be represented"... `hsl(0deg none 50%)` is a valid CSS color that's displayed identically to `hsl(0deg 0% 50%)` but handles interpolation differently.

Reply:
You're correct, I was conflating some unrelated ideas here. I added the logic to allow legacy colors to treat null as missing if space is set in 1e1dae7a819ed0c7d279efc520050edeaf9c32f4. Combined with some refactoring, this diff may be the most helpful- https://github.com/oddbird/sass/compare/e3b8ef726f75679a10ab28ba4713b17e4aa4a7d5...4ada2835df5a714a3901b1bbe140216d20a28eca